radio/src/maths.cpp

```diff
@@ -172,3 +173,30 @@ void getGpsDistance()
     telemetryData.hub.maxGpsDistance = telemetryData.hub.gpsDistance;
 }
 #endif
+
+#if defined(STM32F4)
+uint32_t getCoordDistance(float  y1, float x1, float y2, float x2){
+  float nRadius = 6370997.0; // Earth’s radius in meters
+  float rad = 0.0174532925199433;
+  float nDLat = (y2 - y1) * rad;
+  float nDLon = (x2 - x1) * rad;
+  y1 *= rad;
+  y2 *= rad;
+
+  float nA =   powf( sinf(nDLat/2), 2 ) + cosf(y1) * cosf(y2) * powf( sin(nDLon/2), 2 );
+  float nC = 2 * atan2f( sqrtf(nA), sqrtf( 1 - nA ));
```

Comment:
Careful!  atan2f() can return NaN or Inf if both operands are **close** to zero!